units/generate_tasks_test.go

```diff
@@ -547,6 +638,36 @@ func TestGeneratedTasksAreNotDependencies(t *testing.T) {
 			assert.Equal(foundTask.DependsOn, []task.Dependency{{TaskId: "mci_identifier_testBV2_dependencyTask__01_01_01_00_00_00", Status: evergreen.TaskSucceeded}})
 		}
 	}
+
+	require.NoError(db.ClearCollections(task.Collection, model.ParserProjectCollection))
+
+	// check that the generated tasks are included as dependencies by default
+	pp = model.ParserProject{}
+	err = util.UnmarshalYAMLWithFallback([]byte(shouldGenerateNewBVConfig), &pp)
+	require.NoError(err)
+	pp.Id = "sample_version"
+	require.NoError(pp.Insert())
+	generateTask = task.Task{
+		Id:                    "mci_identifier_generate_tasks_for_version_version_gen__01_01_01_00_00_00",
+		Version:               "sample_version",
+		BuildId:               "b1",
+		Project:               "mci",
+		DisplayName:           "version_gen",
+		GeneratedJSONAsString: sampleGeneratedProject3,
+		Status:                evergreen.TaskStarted,
+	}
+	require.NoError(generateTask.Insert())
+	j = NewGenerateTasksJob(generateTask.Version, generateTask.Id, "1")
+	j.Run(context.Background())
+	assert.NoError(j.Error())
+	tasks = []task.Task{}
+	assert.NoError(db.FindAllQ(task.Collection, db.Query(task.ByVersion("sample_version")), &tasks))
+	assert.Len(tasks, 3)
+	for _, foundTask := range tasks {
```

Comment:
Do we have to check whether the other tasks are activated? Or do they not matter?

Reply:
They're not relevant in this case since before this fix only `testBV4` would be erroneously activated.